### PR TITLE
📖 Move docs to mkdocs-material

### PR DESCRIPTION
**Pull request by ncdc**

### CODE REVIEW

**Review by vincepri — APPROVED** [review]
Thank you for this!

**Comment by ncdc** [comment]
/hold

Will merge tomorrow

File: Makefile [hunk 1]
```diff
@@ -158,9 +158,27 @@ update-contextual-logging: $(LOGCHECK)
 	UPDATE=true ./hack/verify-contextual-logging.sh
 .PHONY: update-contextual-logging
 
-generate-docs:
-	go run hack/generate/cli-doc/gen-cli-doc.go
-	./hack/generate/crd-ref/run-crd-ref-gen.sh
+kcp-docs-image: docs/Dockerfile
+	# Skip if LOCAL is set
+	[[ -n "$$LOCAL" ]] || docker buildx build -f docs/Dockerfile -t kcp-docs --load docs
```
Comment (nrb) [inline Makefile:163]: Why the local option? For quicker iteration?

Comment (ncdc) [reply Makefile:163]: Non containerized

**Pull request merged**